api.php

```diff
@@ -1008,6 +1008,61 @@ public function run($userID, $permissionIsExplicit) {
 	}
 }
 
+/**
+ * API entry game/getmessage
+ */
+class GetMessage extends ApiEntry {
+	public function __construct() {
+		parent::__construct('game/getmessage', 'GET', '', array('gameID','countryID','toCountryID','offset','limit'));
+	}
+	public function run($userID, $permissionIsExplicit) {
+
+		global $DB;
+		$args = $this->getArgs();
+
+		if ($args['gameID'] === null)
+			throw new RequestException('gameID is required.');
+
+		if ($args['countryID'] === null)
+			throw new RequestException('countryID is required.');
+
+		if (intval($args['limit']) > 25)
+			throw new RequestException('limit should not exceed 25');
+
+		$gameID = intval($args['gameID']);
+		$countryID = intval($args['countryID']);
+		$toCountryID = intval($args['toCountryID']);
+		$offset = intval($args['offset']);
+		$limit = intval($args['limit']);
+
+		$limit = $limit ? $limit : 25;
+		$offset =  $offset ? $offset : 0;
+
+		// Global Get all messages addressed to everyone
+		if ( $countryID == 0 ) {
+			$where = "toCountryID = 0";
+		}
+
+		// Only get messages sent between
+		else {
+			$where = "( toCountryID = $toCountryID AND fromCountryID = $countryID )
+					OR
+					( fromCountryID = $toCountryID AND toCountryID = $countryID )";
+		}
```

Comment:
Is there any security risk here such as SQL injection?

Reply:
https://stackoverflow.com/questions/60174/how-can-i-prevent-sql-injection-in-php

Reply:
I looked into this database.php is already using mysqli. I also implemented very similar logic as is in chatbox.php